libraries/launcher/org/prismlauncher/utils/JsonParser.java

```diff
@@ -0,0 +1,425 @@
+ *  GNU General Public License for more details.
+ *
+ *  Linking this library statically or dynamically with other modules is
+ *  making a combined work based on this library. Thus, the terms and
+ *  conditions of the GNU General Public License cover the whole
+ *  combination.
+ *
+ *  As a special exception, the copyright holders of this library give
+ *  you permission to link this library with independent modules to
+ *  produce an executable, regardless of the license terms of these
+ *  independent modules, and to copy and distribute the resulting
+ *  executable under terms of your choice, provided that you also meet,
+ *  for each linked independent module, the terms and conditions of the
+ *  license of that module. An independent module is a module which is
+ *  not derived from or based on this library. If you modify this
+ *  library, you may extend this exception to your version of the
+ *  library, but you are not obliged to do so. If you do not wish to do
+ *  so, delete this exception statement from your version.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * This file incorporates work covered by the following copyright and
+ * permission notice:
+ *
+ *      MIT License
+ *
+ *      Copyright (c) 2022 TheKodeToad
+ *
+ *      Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ *      The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ *      THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.prismlauncher.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.prismlauncher.exception.JsonParseException;
+
+/**
+ * Single-file JSON parser to allow for usage in versions without GSON.
+ */
+public final class JsonParser {
```

Comment:
It would be cleaner and easier to add GSON to the classpath for legacy versions rather than rolling your own. We don't need to reinvent the wheel.

Reply:
I couldn't think of a clean way to add the library depending on the fix being active, and it may conflict with versions used by old mods (I'm unsure on this)?

If it's not removed, this is a parser I've tested extensively.

Reply:
this can be addressed at a later date. it's nearly been a year :sweat_smile:.